src/api/client.ts

```diff
@@ -33,16 +33,31 @@ export class APIError extends Error {
 
     static formatError(error: APIErrorData) {
         if (
-            error.what === 'unspecified' &&
-            error.details[0].file &&
-            error.details[0].file === 'http_plugin.cpp' &&
-            error.details[0].message.slice(0, 11) === 'unknown key'
+            error?.what === 'unspecified' &&
+            error?.details?.[0]?.file === 'http_plugin.cpp' &&
+            error?.details?.[0]?.message?.slice(0, 11) === 'unknown key'
         ) {
             // fix cryptic error messages from nodeos for missing accounts
             return 'Account not found'
-        } else if (error.what === 'unspecified' && error.details && error.details.length > 0) {
+        } else if (error?.what === 'unspecified' && error?.details?.[0]) {
             return error.details[0].message
-        } else if (error.what && error.what.length > 0) {
+        } else if (error?.what?.length > 0 && error?.details?.[0]?.message?.length > 0) {
```

Comment:
Avoid using existential operators (`?`), we have had issues with downstream libraries and applications not being able to import our library due to those and had to scrub them from the codebase

Reply:
No problem, do you have a suggestion on how to proceed? Optional values is the root issue because, detecting undefined | null | '' strings forced the negative tests for code coverage. If the code was restructured to never allow *falsy* values the optionals along with the negative tests could go away. No negative tests, no need for chai.except. Happy to discuss if you want to run through the options. If you know what you want just let me know.

Reply:
Not sure I follow, can't you just expand `?` by doing `(error != undefined && error.what != undefined)` etc?